### PR TITLE
Add Grouping API documentation

### PR DESCRIPTION
**Pull request by coszio**
Docs for qdrant/qdrant#1768

### CODE REVIEW

File: qdrant/v1.2.x/search.md [hunk 1]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+
+{
+    // Same as in the regular search API
+    "vector": [0.2, 0.3, 0.1, 0.5],
+    ...,
+
+    // Grouping parameters
+    "group_by": "document_id",  // Path of the field to group by
+    "limit": 3,                 // Max amount of groups
+    "group_size": 2,             // Max amount of points per group
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:674]:
```suggestion
    "group_size": 2,            // Max amount of points per group
```

File: qdrant/v1.2.x/search.md [hunk 2]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+    collection_name="{collection_name}",
+
+    # Same as in the regular search() API
+    vector=[0.2, 0.3, 0.1, 0.5],
+    ...,
+    
+    # Grouping parameters
+    group_by="document_id", # Path of the field to group by
+    limit=3,                # Max amount of groups
+    group_size=2,            # Max amount of points per group
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:689]:
```suggestion
    group_size=2,           # Max amount of points per group
```

File: qdrant/v1.2.x/search.md [hunk 3]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+{
+    // Same as in the regular recommend API
+    "negative": [1],
+    "positive": [2, 5],
+    ...,
+
+    // Grouping parameters
+    "group_by": "document_id",  // Path of the field to group by
+    "limit": 3,                 // Max amount of groups
+    "group_size": 2,             // Max amount of points per group
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:707]:
```suggestion
    "group_size": 2,            // Max amount of points per group
```

File: qdrant/v1.2.x/search.md [hunk 4]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+
+    # Same as in the regular recommend() API
+    negative=[1],
+    positive=[2, 5],
+    ...,
+
+    # Grouping parameters
+    group_by="document_id", # Path of the field to group by
+    limit=3,                # Max amount of groups
+    group_size=2,            # Max amount of points per group
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:723]:
```suggestion
    group_size=2,           # Max amount of points per group
```

File: qdrant/v1.2.x/search.md [hunk 5]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+                ]
+            }
+        ],
+    },
+    "status": "ok",
+    "time": 0.001
+}
+```
+
+The groups will be ordered by the score of the top point in the group. Inside each group the points will be sorted too.
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:760]: Maybe it's better to use _are_ versus _will be_.

```suggestion
The groups are ordered by the score of the top point in the group. Inside each group the points are sorted too.
```

File: qdrant/v1.2.x/search.md [hunk 6]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+*Available since v1.2.0*
+
+It is possible to group results by a certain field. This is useful when you have multiple points for the same item, and you want to avoid redundancy of the same item in the results.
+
+For example, if you have a large document split into multiple chunks, and you want to search or recommend on a per-document basis, you can group the results by the document ID. 
+
+Consider having points with the following payloads:
+
+```json
+{
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:607]: Maybe we should add some dummy vectors, just with one dimension, to at least make the JSON object valid.

**Review by timvisee — APPROVED** [review]
Nice work!

File: qdrant/v1.2.x/search.md [hunk 7]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+    {
+        "id": 5,
+        "payload": {
+            "chunk_part": 0, 
+            "document_id": 2.42,
+        },
+        "vector": ...,
+    },
+}
+```
```
Comment (timvisee) [inline qdrant/v1.2.x/search.md:657]: Maybe we should add some dummy vectors, just with one dimension, to at least make the JSON object valid.

**Review by davidmyriel — APPROVED** [review]
LGTM

File: qdrant/v1.2.x/search.md [hunk 8]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+            "chunk_part": 1, 
+            "document_id": "a",
+        },
+        "vector": ...,
+    },
+    {
+        "id": 2,
+        "payload": {
+            "chunk_part": 2, 
+            "document_id": "a",
```
Comment (agourlay) [inline qdrant/v1.2.x/search.md:628]: what do you think about adding an array example right away here?

File: qdrant/v1.2.x/search.md [hunk 9]
```diff
@@ -592,3 +592,176 @@ It is impossible to retrieve Nth closest vector without retrieving the first N v
...
+        "payload": {
+            "chunk_part": 0, 
+            "document_id": 2.42,
+        },
+        "vector": ...,
+    },
+}
+```
+
+With the ***groups*** API, you will be able to get the top *N* points for each document, assuming that the payload of the points contains the document ID.
```
Comment (agourlay) [inline qdrant/v1.2.x/search.md:659]: Are we returning only full groups OR can we respond with groups with less then N elements?

Comment (agourlay) [reply qdrant/v1.2.x/search.md:659]: I just realized you have an example afterwards with a group that is not full :+1: 

Wondering if it is worth it to more precise on the grouping semantic.

File: qdrant/v1.2.x/search.md (hunk 7 above)
Comment (coszio) [reply qdrant/v1.2.x/search.md]: Added 👍

**Comment by coszio** [comment]
I am happy with this, merging

**Pull request merged**